packages/artifact/src/internal/download/download-artifact.ts

```
@@ -140,13 +144,16 @@ export async function downloadArtifactPublic(
 
   try {
     core.info(`Starting download of artifact to: ${downloadPath}`)
-    await streamExtract(location, downloadPath)
-    core.info(`Artifact download completed successfully.`)
+    if (await streamExtract(location, downloadPath)) {
+      core.info(`Artifact download completed successfully.`)
+      return {downloadPath, skipped: false}
+    } else {
+      core.info(`Artifact download skipped.`)
```

If a user requests a specific artifact to download and that artifact can't be downloaded, why should that silently fail?

Shouldn't that be an error?

---

I'm reluctant to change the default behavior here as it could be a breaking change for users.

---

iiuc the API only expects zip content type for extraction when used with `actions/download-artifact@v4`: https://github.com/actions/toolkit/blob/bb2278e5cfbb40afc20890c415e9ffa836631cd5/packages/artifact/src/internal/download/download-artifact.ts#L92

And it [currently breaks](https://github.com/actions/download-artifact/issues/328#issuecomment-2194122935) workflows when artifacts with other content-type are being downloaded.

I guess we could create an error type is content type does not match so `actions/download-artifact` can catch it?

---

I thought silently skipping unrelated artifacts not uploaded with `actions/upload-artifact` would be best so it doesn't require any changes in `actions/download-artifact`.

---

Would it be better to use `pattern` [input](https://github.com/actions/download-artifact?tab=readme-ov-file#inputs) to filter out any artifacts that are not created by `upload-artifact` action in this case?  (credit to @joshmgross for this suggestion as well!)

---

@yacaovsnc `pattern` is optional afaik and would therefore any artifact created in the workflow

---

@crazy-max yeah I understand this is an issue with non-zip files so we are hoping to mitigate this issue with existing tools.  Maybe an exclude pattern such as `!**/*.gzip` if all docker uploaded artifacts are gzips?  Of if the artifacts have a common name pattern, it can also be matched? Is there a pattern for docker uploaded artifacts? 

Also, if there are customer who rely on download docker uploaded artifacts, we are breaking them either way.  Would it be possible to actually upload zips? I understand this maybe difficult too.

---

> What's wrong with excluding artifacts that don't have the expected content-type during extraction? If someone else using the API to upload another content-type, people using actions/download-artifact would have the same issue.

I think the issue is it blurs the line between a real exception and an "expected" exception?  Consider the same scenario where a user uses API to upload, and uses download-artifact to verify the content.  Everything could be working until someone accidentally modify the upload portion to point to a wrong file, with this change it will be hard for those users to catch the mistake?

`pattern` exists to include and exclude files, and to me it fits the current situation and it could solve our issue without any code change?  A exclude pattern of `!**/*.dockerbuild` seems could do the job?

---

> So what you mean is having this pattern set as default in actions/download-artifact?

No, customers must manually update their workflow to specify the exclusion pattern.  We need to provide guidance on how to properly configure the download step to skip over docker produced artifacts.

---

> So I'm not sure why this would be expected. I think this is an oversight in the implementation to skip unsupported content-types.

I suspect the disagreement here is fundamentally - "should the 1st party artifact actions be compatible with artifacts created by other actions?"

The current implementation says no - thus users should avoid downloading artifacts that weren't uploaded by `actions/upload-artifact`.

The behavior change to skip non-zip files does not make them *compatible*, but it would at least avoid some friction due to incompatibility.

I'm still not convinced skipping these downloads is the right approach though - changing an explicit failure to a silent one could surprise users and increase friction.